tensorflow_io/core/kernels/azfs_kernels.cc

```diff
@@ -356,249 +454,235 @@ class AzBlobWritableFile : public WritableFile {
-                            ")");
+  ParseAzBlobPath(path, false, &account, &container, &object, status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
   }
+  file->plugin_file = new AzBlobWritableFile(account, container, object);
 
-  return Status::OK();
+  TF_SetStatus(status, TF_OK, "");
+}
+
+static void NewReadOnlyMemoryRegionFromFile(const TF_Filesystem* filesystem,
+                                            const char* path,
+                                            TF_ReadOnlyMemoryRegion* region,
+                                            TF_Status* status) {
+  TF_SetStatus(status, TF_UNIMPLEMENTED,
+               "NewReadOnlyMemoryRegionFromFile not implemented");
 }
 
-Status AzBlobFileSystem::CreateDir(const std::string &dirname) {
+static void CreateDir(const TF_Filesystem* filesystem, const char* path,
+                      TF_Status* status) {
   std::string account, container, object;
-  TF_RETURN_IF_ERROR(
-      ParseAzBlobPath(dirname, true, &account, &container, &object));
+  ParseAzBlobPath(path, true, &account, &container, &object, status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
   if (container.empty()) {
-    return errors::FailedPrecondition("Cannot create storage accounts");
+    TF_SetStatus(status, TF_FAILED_PRECONDITION,
+                 "Cannot create storage accounts");
+    return;
   }
 
   // Blob storage has virtual folders. We can make sure the container exists
   auto blob_client_wrapper = CreateAzBlobClientWrapper(account);
 
   if (blob_client_wrapper.container_exists(container)) {
-    return Status::OK();
+    TF_SetStatus(status, TF_OK, "");
+    return;
   }
 
   blob_client_wrapper.create_container(container);
   if (errno != 0) {
-    return errors::Internal("Failed to create directory ", dirname, " (",
-                            errno_to_string(), ")");
+    std::string error_message = absl::StrCat(
+        "Failed to create directory ", path, " (", errno_to_string(), ")");
+    TF_SetStatus(status, TF_INTERNAL, error_message.c_str());
+    return;
   }
-  return Status::OK();
+  TF_SetStatus(status, TF_OK, "");
+}
+
+static void RecursivelyCreateDir(const TF_Filesystem* filesystem,
+                                 const char* path, TF_Status* status) {
+  CreateDir(filesystem, path, status);
 }
 
-Status AzBlobFileSystem::DeleteDir(const std::string &dirname) {
+static void DeleteFile(const TF_Filesystem* filesystem, const char* path,
+                       TF_Status* status) {
+  std::string account, container, object;
+  ParseAzBlobPath(path, false, &account, &container, &object, status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
+
+  auto blob_client = CreateAzBlobClientWrapper(account);
+
+  blob_client.delete_blob(container, object);
+  if (errno != 0) {
+    std::string error_message =
+        absl::StrCat("Failed to delete ", path, " (", errno_to_string(), ")");
```

Comment:
append "ERROR_CODE" before `errno_to_string()` for clarity purposes.

Reply:
@kvignesh1420 Updated.